src/klein/_app.py

```diff
@@ -39,17 +39,41 @@
 from ._decorators import modified, named
 from ._interfaces import IKleinRequest
 from ._resource import KleinResource
-from ._typing import Awaitable, KwArg, VarArg
 
 
 KleinSynchronousRenderable = Union[str, bytes, IResource, IRenderable]
 KleinRenderable = Union[
     KleinSynchronousRenderable, Awaitable[KleinSynchronousRenderable]
 ]
-KleinRoute = Callable[[Any, IRequest, VarArg(Any), KwArg(Any)], KleinRenderable]
-KleinErrorHandler = Callable[
-    [Optional["Klein"], IRequest, Failure], KleinRenderable
-]
+
+
+# Protocol is new in Python 3.8
+try:
+    from typing import Protocol
+except ImportError:
+    KleinRoute = Callable[..., KleinRenderable]
+    KleinErrorHandler = Callable[..., KleinRenderable]
```

Comment:
This is wimpier than the prior spec but when we do type checking, we use Python 3.8+, so the Protocol below.